### ISSUE TITLE
Two running instances of Mocha in Watch mode hang on first change detection

### ISSUE THREAD

**Issue by FL0RIANMEYER** [mochajs/mocha#3404]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend avoiding the use of globally installed Mocha.

### Description
Two running instances of Mocha in Watch mode hang on first change detection. Startup execution work as expected in both instances. Both instances just print 0 passing without failure prints. No Global instance of Mocha installed. Both instances in independent node_module directories.

### Steps to Reproduce
_package.json:_
```
"scripts": {
    "test:devTest:service": "mocha -w compiled/devTest/test/unit/**/*.js -r mock-css-modules",
    "test:devTest:system": "mocha -w ../compiled/test/e2e/**/*.js",
}
```
_shell:_
```
npm run test:devTest:service
npm run test:devTest:system // in second terminal
```

**Expected behavior:** [What you expect to happen]
Execute both instances as expected

**Actual behavior:** [What actually happens]
Two running instances of Mocha in Watch mode hang on first change detection.
Both instances just print 0 passing without failure prints.
**Reproduces how often:** [What percentage of the time does it reproduce?]

### Versions

- The output of `mocha --version` and `node node_modules/.bin/mocha --version`:
5.2.0
- The output of `node --version`:
v9.4.0
- The version and architecture of your operating system:
Windows 7 64

**Comment by outsideris** [comment]
On macOS, it is fine. I should check it on Windows.

Could you provide [mcve](https://stackoverflow.com/help/mcve)?

**Comment by FL0RIANMEYER** [comment]
The reason seems to be the massive change injection of [webpack-emit-all-plugin](https://github.com/DrewML/webpack-emit-all-plugin).

I think mocha can't handle a lot of changes in a short time. With two instances the problem occure just faster. **Same with one instance** but after the n-th compilation.

**1.** Just try setting up a slow test suite so that Mocha is killed after the first changes due to further changes.
**2.** Make many changes in a short time to the observed directory.

**1.** 
```javascript
it('t', async function(done) {
   this.timeout(2500);
   setTimeout(() => done, 2000);
});
```

**2.** 
```javascript
const fs = require('fs');

let i = 0;
setInterval(() => {
   fs.writeFile('./test.js', `const a = ${i++}`, () => {});
}, 10); // <- Adjust if too aggressive

setTimeout(() => process.exit(0), 3000);
```

or maybe
```javascript
fs.writeFile(`./test${i}.js`, `const a = ${i++}`, () => {});
```

### Solution 

- One solution could be to add a delay parameter to the -w argument _-w 1000_. But this could also create a flaky test behavior as the project grows. Or unnecessary long wait before testing.

- An alternative could be to open an HTTP port to trigger events sent by other programs such as Webpack or Babel.

This problem can not be associated only with Webpack. The same behavior could also occur with babel or other compilers that produce many changes in the observed directory.

**Comment by plroebuck** [comment]
Looks like you're generating files to be tested. Did you attempt to make use of `--delay`?

**Comment by juergba** [comment]
The watch feature has been  refactored completely using `chokidar`. If problem persists, please open a new issue. Closed by #3980.
